File: Load Balancer.java
import java.net.*;
import java.io.*;

public class loadbalancer{
	public static void main(String [] args)throws IOException
	{	
	ServerSocket s=new ServerSocket(7777);
	while(true)
	{
	Socket connectionsocket=s.accept();
	BufferedReader messagefromclient = new BufferedReader(new InputStreamReader(connectionsocket.getInputStream())):
	String clientsentence=messagefromclient.readLine();
	
	if(clientsentence.contains("Hello")
	{
	Socket server=new Socket("192.168.159.128" ,3333);
	DataOutputStream outtoserver=new DataOutputStream(server.getOutputStream());
	outtoserver.writeBytes(clientsentnce+"\n");
	server.close();
	}

	else if(clientsentnce.contains("Hai"))
	{
	Socket server1=new Socket("912.168.159.130" ,8888);
	DataOutputStream outtoserver=new DataOutputStream(server1.getOutputStream());
	outtoserver.writeBytes(clientsentece+"\n");
	server1.close();
	}

	else
	{
	System.out.println("ERROR!!");
	}
	
	}
	}
}

File: Load Balancer.rb
require 'socket'

loadbalancer=TCPServer.open("912.168.35.129",3333)

while true
client=loadbalancer.accept

rhost=client.peeraddr.last
puts"#{rhost} connect to loadbalancer"

fromclient=client.gets
puts "Message From the client: *{fromclient}"

if fromclient=="Hello"
sockettoserver=TCPSocket.open("192.168.159.128",3030)
sockettoserver.write(fromclient)
sockettoserver.close

if fromclient=="Hai"
sockettoserver1=TCPSocket.open("192.168.159.130",4040)
sockettoserver1.write(fromclient)
sockettoserver1.close

elso
puts "ERROR"

end
end
end
